Make test scripts run without /bin/bash

twz123 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Make sure to read the Contributing Guide before submitting your PR: https://github.com/helm/helm/blob/main/CONTRIBUTING.md
2. If this PR closes another issue, add 'closes #<issue number>' somewhere in the PR summary. GitHub will automatically close that issue when this PR gets merged. Alternatively, adding 'refs #<issue number>' will not close the issue, but help provide the reviewer more context.-->

**What this PR does / why we need it**:

The test scripts hardcoded `#!/bin/bash` while they are not really requiring bash. Use the more portable `#!/usr/bin/env sh` instead, so that they use the default shell. This makes things work on systems without a bash in the /bin folder.

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

gjenkins8 commented:
@twz123 -- can you please sign your commits: https://github.com/helm/helm/pull/11991/checks (more details: https://helm.sh/blog/helm-dco/index.html)

I presume (as a community member) if the systems `sh` suffices to run tests, then this is a welcome change. Though, I don't know of any (modern) OS which doesn't supply `/bin/bash`. And generally `bash` is more featureful than what posix shell compatibility (`sh`). Not sure if anyone will have strong opinions in limiting functionality to `sh` here.

It might also be worth detailing the usecase for strictly adhering to `sh`?

twz123 committed:
Make test scripts run without /bin/bash

The test scripts hardcoded #!/bin/bash while they are not really
requiring bash. Use the more portable #!/usr/bin/env sh instead, so that
they use the default shell.

Signed-off-by: Tom Wieczorek <twieczorek@mirantis.com>

benoittgt commented:
Based on "Some distros don't ship bash by default, most notably Alpine. Since those scripts are by all means POSIX compatible, it seems unnecessary to require bash"

I think we should validate this PR.

TerryHowe reviewed (approved):
/lgtm
